src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -225,6 +227,10 @@ def upload_file(
                 raise ne.with_traceback(tb) from None
         else:
             uri = normalize_local_file(path)
+            if SNOWFLAKE_CONNECTOR_VERSION >= (3, 0, 3, None):
+                kwargs = {"_skip_upload_on_content_match": skip_upload_on_content_match}
```

Comment:
I chose to do it this way because I can just send `kwargs` to the `run_query` below using `**kwargs`. Otherwise, I would write the `run_query` with arguments twice

Reply:
I see, that makes sense. I guess if we decide to bump the minimum python connector dependency, this could be avoided 🤔